worlds/stick_ranger/Rules.py

```diff
@@ -0,0 +1,221 @@
+from typing import TYPE_CHECKING, Callable, Dict, List
+
+from BaseClasses import CollectionState, MultiWorld
+from worlds.generic.Rules import set_rule
+
+from .Constants import GOAL_EXIT_LOCATIONS, RANGER_CLASSES
+from .Items import unlocks_by_region
+from .Options import SROptions
+
+if TYPE_CHECKING:
+    from . import StickRanger
+
+
+def class_count(state, player: int) -> int:
+    """Return the number of ranger classes the player has unlocked."""
+    return sum(state.has(f"Unlock {cls} Class", player) for cls in RANGER_CLASSES)
```

Comment:
Would recommend putting the classes in an item name group, then just using `state.has_group` or similar.